### ISSUE TITLE
 Boxy SVG not installing

### ISSUE THREAD

**Issue by Rak1ta** [Botspot/pi-apps#2237]
### Confirmations

- [X] I searched for similar bug reports in https://github.com/Botspot/pi-apps/issues and found none were relevant.
- [X] I tried updating Pi Apps and the issue is still not fixed. [How to update Pi Apps?](https://github.com/Botspot/pi-apps#to-update-pi-apps)
- [X] IMPORTANT: I promise to respond to comments on this issue and write a good bug report https://www.softwaretestinghelp.com/how-to-write-good-bug-report/

### What happened?

Error during installation

### Description

I'm getting an error with the latest changes.
[install-fail-Boxy SVG.log](https://github.com/Botspot/pi-apps/files/10437382/install-fail-Boxy.SVG.log)
[2023-01-17T16_38_43_962Z-debug-0.log](https://github.com/Botspot/pi-apps/files/10437383/2023-01-17T16_38_43_962Z-debug-0.log)

If you use this [commit](https://github.com/Botspot/pi-apps/commit/1aec7c43a060fa93dd45dbb4fee2172594dfb08a), then the installation occurs correctly.
[install-success-Boxy SVG.log](https://github.com/Botspot/pi-apps/files/10437689/install-success-Boxy.SVG.log)

### What are your system specs (run the following command in your terminal)?

```bash
OS: Raspbian GNU/Linux 10 (buster)
OS architecture: 32-bit
Last updated Pi-Apps on: 01/16/2023
Latest Pi-Apps version: 01/17/2023
Kernel: armv7l 5.10.103-v7l+
Device model: Raspberry Pi 4 Model B Rev 1.2 
Cpu name: Cortex-A72
Ram size: 3.65 GB
Raspberry Pi OS image version: 2021-05-07
Language: en_US.UTF-8
```

**Comment by Botspot** [comment]
I'm just curious: were you presented with the option to send this in as an error report?
The reason I ask is that I don't see this reported in our `error-logs` Discord channel, and I am wondering if this has only affected you or if it has happened to others.

**Comment by Rak1ta** [comment]
@Botspot
If I update multiple apps at once (in my case, Boxy SVG was the first on the list) then pi-apps didn't offer to send a report.  When I separately reinstalled, the report was sent.  I hope Google translated the text correctly.

**Comment by Botspot** [comment]
I was able to repeat this error in my Buster VM. Installing `libfontconfig-dev` fixed it, so in commit https://github.com/Botspot/pi-apps/commit/f5ca9205f31548bbe9a74949d335571c354d8b07, I made sure that package was installed.

**Comment by Rak1ta** [comment]
@Botspot The fix didn't work.
[install-fail-Boxy SVG.log](https://github.com/Botspot/pi-apps/files/10467624/install-fail-Boxy.SVG.log)
[2023-01-20T15_06_38_808Z-debug-0.log](https://github.com/Botspot/pi-apps/files/10467625/2023-01-20T15_06_38_808Z-debug-0.log)

**Comment by Botspot** [comment]
> Returned to the working scenario of the script.

That version of script was fairly hacky and could need further changes in the future if the Boxy SVG developer changes his build process. The current script uses the exact same build process as the official Snap builds, so I want to avoid downgrading it for everyone.

I have not been able to reproduce the error under the current script version on Buster.
In combination with most reports of openssl_gyp going away after early 2022, I am beginning to think that this problem occurs because something is out of date on your system.

@rak1ta, Could you try updating node packages?
```
npm update -g
```
Also please find out what version of node you have installed, by sending me the output of this command:
```
nvm ls
```

**Comment by Rak1ta** [comment]
@Botspot currently installed
```
pi@raspberrypi:~ $ nvm ls
->      v17.3.0
         system
default -> node (-> v17.3.0)
iojs -> N/A (default)
unstable -> N/A (default)
node -> stable (-> v17.3.0) (default)
stable -> 17.3 (-> v17.3.0) (default)
lts/* -> lts/gallium (-> N/A)
lts/argon -> v4.9.1 (-> N/A)
lts/boron -> v6.17.1 (-> N/A)
lts/carbon -> v8.17.0 (-> N/A)
lts/dubnium -> v10.24.1 (-> N/A)
lts/erbium -> v12.22.8 (-> N/A)
lts/fermium -> v14.18.2 (-> N/A)
lts/gallium -> v16.13.1 (-> N/A)

```

**Comment by Botspot** [comment]
Current version of Node is v19.4.0. Updating to that may help. I was under the impression that NVM auto-updated node, but that does not seem to be the case. (Will fix this from Pi-Apps' end)

For now @Rak1ta, please update node with this command:
```
nvm install node
nvm install-latest-npm
```
And then try to install Boxy SVG with the latest Pi-Apps script.

**Comment by Rak1ta** [comment]
@Botspot 
```
pi@raspberrypi:~ $ npm update -g
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'npm@9.3.1',
npm WARN EBADENGINE   required: { node: '^14.17.0 || ^16.13.0 || >=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v17.3.0', npm: '8.3.0' }
npm WARN EBADENGINE }

removed 39 packages, changed 272 packages, and audited 403 packages in 25s

42 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
pi@raspberrypi:~ $ npm fund
npm WARN cli npm v9.3.1 does not support Node.js v17.3.0. You should probably upgrade to a
npm WARN cli newer version of node as we can't make any promises that npm will work with this
npm WARN cli version. This version of npm supports the following node versions: `^14.17.0 ||
npm WARN cli ^16.13.0 || >=18.0.0`. You can find the latest version at https://nodejs.org/.
pi

```

**Comment by Rak1ta** [comment]
@Botspot After the update it worked.

**Comment by Botspot** [comment]
Good! I have just [uploaded](https://github.com/Botspot/pi-apps/commit/0de0e625f718e321b40dd31d10d519fcfac75c73) improvements to our Node.js script that should keep Node up-to-date.

**Comment by Rak1ta** [comment]
> but that does not seem to be the case

Is the [app description](https://github.com/Botspot/pi-apps/blob/master/apps/Node.js/description) up to date?

**Comment by Botspot** [comment]
> > but that does not seem to be the case
> 
> Is the [app description](https://github.com/Botspot/pi-apps/blob/master/apps/Node.js/description) up to date?

I must have been mistaken when I wrote in the description that nvm automatically keeps node up to date. But that claim is now true, as of today. :)

> @Botspot I would like to clarify on updating Node.js. So it should be that nothing refers to [this version](https://github.com/Botspot/pi-apps/blob/873bdceceea0457cbce6ef5527a3a9eb495bb8f1/apps/Node.js/install#L3)? Or is it for version reference?

This takes full advantage of our Github bot which periodically checks for new versions of apps. I added this script today: https://github.com/Botspot/pi-apps/blob/master/.github/workflows/updates/Node.js.sh
This script is used to check for the current version of node.js and nvm, then it compares those values to the ones found in the script, denoted as `version=` and `version2=`.
If there is a newer version than what was spotted in the script, then our github bot will open a Pull Request with the necessary changes. See: https://github.com/Botspot/pi-apps/pull/2244

You are correct that the $version variable is not actually used anywhere in the app's `install` script. But when that value is updated on Github, it will **trigger a Node.js reinstall** for all Pi-Apps users, thereby keeping it up to date.
